exporter/exporter.go

```diff
@@ -25,119 +21,29 @@ type cat1data struct {
-		if dataCriteria.Value.Type == "CD" {
-			mappedDc.ResultOids = append(mappedDc.ResultOids, dataCriteria.CodeListID)
-		}
-
-		if dc.DataCriteriaOid != "" {
-			mappedDataCriteria[dc] = mappedDc
-		}
-	}
-
-	// Add the key to the value to get what HDS would have returned
-	var retDataCriteria []mdc
-	for key, value := range mappedDataCriteria {
-		value.DataCriteriaOid = key.DataCriteriaOid
-		value.ValueSetOid = key.ValueSetOid
-		retDataCriteria = append(retDataCriteria, value)
-	}
-	return retDataCriteria
-}
-
-func exporterFuncMap(cat1Template *template.Template) template.FuncMap {
+func exporterFuncMap(cat1Template *template.Template, vsMap models.ValueSetMap) template.FuncMap {
 	return template.FuncMap{
-		"timeNow":                      time.Now().UTC().Unix,
-		"newRandom":                    uuid.NewRandom,
-		"timeToFormat":                 timeToFormat,
-		"identifierForInt":             identifierForInt,
-		"identifierForString":          identifierForString,
-		"escape":                       escape,
-		"entryInfosForPatient":         entryInfosForPatient,
-		"executeTemplateForEntry":      generateExecuteTemplateForEntry(cat1Template),
-		"condAssign":                   condAssign,
-		"valueOrNullFlavor":            valueOrNullFlavor,
-		"dischargeDispositionDisplay":  dischargeDispositionDisplay,
-		"sdtcValueSetAttribute":        sdtcValueSetAttribute,
-		"getTransferOid":               getTransferOid,
-		"identifierForInterface":       identifierForInterface,
-		"codeToDisplay":                codeToDisplay,
-		"valueOrDefault":               valueOrDefault,
-		"reasonValueSetOid":            reasonValueSetOid,
-		"oidForCodeSystem":             oidForCodeSystem,
-		"oidForCode":                   oidForCode,
+		"timeNow":             time.Now().UTC().Unix,
+		"newRandom":           uuid.NewRandom,
+		"timeToFormat":        timeToFormat,
+		"identifierForInt":    identifierForInt,
+		"identifierForString": identifierForString,
+		"escape":              escape,
+		// mattrbianchi: I searched for this and it wasn't used in any of the templates... should it really be in here?
+		//"entryInfosForPatient":         models.Record.EntryInfosForPatient,
+		"executeTemplateForEntry":     generateExecuteTemplateForEntry(cat1Template),
+		"condAssign":                  condAssign,
+		"valueOrNullFlavor":           valueOrNullFlavor,
+		"dischargeDispositionDisplay": dischargeDispositionDisplay,
+		"sdtcValueSetAttribute":       sdtcValueSetAttribute,
+		"getTransferOid":              getTransferOid,
+		"identifierForInterface":      identifierForInterface,
+		"codeToDisplay":               codeToDisplay,
+		"valueOrDefault":              valueOrDefault,
+		"reasonValueSetOid":           vsMap.ReasonValueSetOid,
+		"oidForCodeSystem":            oidForCodeSystem,
+		// This is also not used in the templates
+		"oidForCode":                   vsMap.OidForCode,
```

Comment:
This is used in at least one of the templates; `_2.16.840.1.113883.10.20.24.3.7.xml`.